devito/profiler.py

```diff
@@ -1,30 +1,86 @@
-from ctypes import Structure, byref, c_double
+from ctypes import Structure, byref, c_double, c_longlong
 
-from cgen_wrapper import Block, Statement, Struct, Value
+from cgen_wrapper import Assign, Block, For, Statement, Struct, Value
+from devito.logger import error
 
 
-class Profiler(Structure):
+class Profiler(object):
     """The Profiler class is used to manage profiling information for Devito
     generated C code.
     """
+    TIME = 1
+    FLOP = 2
 
-    def __init__(self):
-        self.name = "timings"
-        self.fields = []
-        self.timings = None
+    def __init__(self, openmp=False):
+        self.openmp = openmp
+        self.t_name = "timings"
+        self.o_name = "oi"
+        self.f_name = "flops"
+        self.loop_temp_prefix = "temp_"
```

Comment:
Maybe ``_C_timings``, ``_C_oi``, ``_C_flops``, ``_C_temporary``. Plus add a comment that says that attributes starting with ``_C_`` are simply used for code generation ?